app/admin/teams.rb

```diff
@@ -0,0 +1,328 @@
+include ActiveAdminHelpers
+
+ActiveAdmin.register Team do
+  permit_params :name, user_ids: []
+
+  form do |f|
+    f.inputs 'Team Details' do
+      f.input :name
+      f.input :users, as: :check_boxes, collection: User.all.map { |user| [user.email, user.id] }
+    end
+    f.actions
+  end
+
+  action_item :import_csv, only: :index do
+    link_to 'Import CSV', import_csv_admin_teams_path
+  end
+
+  collection_action :import_csv, method: [:get, :post] do
+    if request.post?
+      file = params[:file].tempfile
+      options = { header_transformations: [:downcase], col_sep: ',', row_sep: :auto }
+      csv_data = SmarterCSV.process(file, options)
+
+      csv_data.each do |row|
+        team = Team.find_or_create_by(name: row[:name])
+        
+        if row[:user_emails].nil?
+          user_emails = []
+        else
+          user_emails = row[:user_emails].split(',')
+        end
+
+        existing_user_ids = team.users.pluck(:id)
+
+        user_emails.each do |email|
+          user = User.find_by(email: email.strip.downcase)
+          if user
+            UsersTeam.find_or_create_by(user_id: user.id, team_id: team.id)
+            existing_user_ids.delete(user.id)
+          end
+        end
+
+        team.users.delete(User.where(id: existing_user_ids))
```

Comment:
I think it's better remove all `team.users.destroy_all` in 33 line and then you don't need any manipulate with this field in 39 and 43 lines

Reply:
team.users.destroy_all: This would destroy all the User records associated with the team. Not only would the users be removed from the team, but the User records themselves would be deleted from the database. 

team.users.delete(User.where(id: existing_user_ids)): This removes only the associations between the users and the team. The User records themselves are not deleted from the database. They are just no longer associated with the particular team. This is the safer operation as it maintains the integrity of our data.

Our goal is to only update the users for a team based on the new CSV file without deleting any User records, then team.users.delete(User.where(id: existing_user_ids)) is the better choice.

Reply:
let's discuss it today